pkg/archive/archive_linux.go

```diff
@@ -236,7 +238,9 @@ func createDirWithOverlayOpaque(tmp string) (string, error) {
 	if err := os.MkdirAll(lowerDummy, 0700); err != nil {
 		return "", errors.Wrapf(err, "failed to create a dummy lower directory %s", lowerDummy)
 	}
-	mOpts := fmt.Sprintf("lowerdir=%s,upperdir=%s,workdir=%s", lower, upper, work)
+	// lowerdir needs ":" to be escaped: https://github.com/moby/moby/issues/40939#issuecomment-627098286
+	lowerEscaped := strings.ReplaceAll(lower, ":", "\\:")
+	mOpts := fmt.Sprintf("lowerdir=%s,upperdir=%s,workdir=%s", lowerEscaped, upper, work)
```

Comment:
Does the same apply  to `upper` and `work`?

Wondering if we should have a `clean(path string)` function to handle this, and then;

```suggestion
	mOpts := fmt.Sprintf("lowerdir=%s,upperdir=%s,workdir=%s", clean(lower), clean(upper), clean(work))
```

Or perhaps do it in one go, something like;

```suggestion
	mOpts := fmt.Sprintf("lowerdir=%s,upperdir=%s,workdir=%s", lowerEscaped, upper, work)
	mOpts = clean(mOpts)
```

Reply:
Specific to lowerdir (because it can be comma-sepatated list)